sql/src/main/java/io/crate/analyze/AnalyzedUpdateStatement.java

```diff
@@ -59,8 +75,14 @@ public Symbol query() {
         return query;
     }
 
-    public Multimap<String, Symbol> returningClause() {
-        return returningClause;
+    @Nullable
```

Comment:
I still don't feel good about the `Nullable` here + `ResultSetStatement` solution.

The interface doesn't mark `fields()` as `Nullable`, so whoever uses a `resultSetStatement` might assume that it is not-null, but then we've specific implementations like this where `fields` is nullable.

This is rather trappy imho and a NPE waiting to happen.
In https://github.com/crate/crate/pull/9481/files#diff-8df7e31e8934a6bebea6f7bf4532b50cR596-R599 it is taking care of it, but still.

How about we move `fields()` to `AnalyzedStatement`, make it `Nullable` and get rid of the `ResultSetStatement`  interface? If the fields are there, it is a statement with result set, if it is not there, it isn't.

Reply:
Sounds simple and pragmatic 👍